### PR TITLE
AIP-84: Migrating DELETE a queued asset events for DAG to fastAPI

### PR DESCRIPTION
**Pull request by amoghrajesh**
related: https://github.com/apache/airflow/issues/42370
Migrating delete a queued asset events for DAG to fastAPI

Dependent on https://github.com/apache/airflow/pull/43934

Same setup as https://github.com/apache/airflow/pull/43934

Responses:
1. Legacy
image

2. FastAPI
image

With time filtering:
1. Legacy
image

2. FastAPI
image

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by amoghrajesh** [comment]
Only last 2 commits are relevant

File: airflow/api_fastapi/core_api/routes/public/dags.py [hunk 1]
```diff
@@ -303,3 +306,76 @@ def delete_dag(
...
+    "/{dag_id}/assets/queuedEvent/{uri:path}",
+    status_code=status.HTTP_204_NO_CONTENT,
+    responses=create_openapi_http_exception_doc(
+        [
+            status.HTTP_400_BAD_REQUEST,
+            status.HTTP_404_NOT_FOUND,
+        ]
+    ),
+)
+def delete_dag_asset_queued_event(
```
Comment (pierrejeambrun) [inline airflow/api_fastapi/core_api/routes/public/dags.py:365]: Should go in asset endpoint.

File: airflow/api_fastapi/core_api/routes/public/dags.py [hunk 2]
```diff
@@ -303,3 +306,76 @@ def delete_dag(
...
+    dag_id: str,
+    uri: str,
+    session: Annotated[Session, Depends(get_session)],
+    before: str = Query(None),
```
Comment (pierrejeambrun) [inline airflow/api_fastapi/core_api/routes/public/dags.py:369]: Use the `DateTimeQuery`, validation and parsing of parameters should be done by the route. So downstream functions manipulated a validated input, and not a bare string.

**Review by pierrejeambrun — COMMENTED** [review]
Nice. A few suggestions, more in depth review when the base branch is merged and this is rebased. (but it's already close to mergeable I would say)

File: airflow/api_fastapi/core_api/datamodels/assets.py [hunk 3]
```diff
@@ -99,3 +99,18 @@ class AssetEventCollectionResponse(BaseModel):
 
     asset_events: list[AssetEventResponse]
     total_entries: int
+
+
+class QueuedEventResponse(BaseModel):
+    """Queued Event serializer for responses.."""
```
Comment (rawwar) [inline airflow/api_fastapi/core_api/datamodels/assets.py:105]:
```suggestion
    """Queued Event serializer for responses."""
```

**Pull request closed without merging**